src/stactools/core/utils/raster_footprint.py

```diff
@@ -278,38 +802,42 @@ def densify_reproject_simplify(
 ) -> Polygon:
     """
-    From the input Polygon, densifies the polygon, reprojects it to EPSG:4326, and then
-    simplifies the resulting polygon.
+    Densifies the input polygon, reprojects it to EPSG 4326, and simplifies the
+    resulting polygon.
 
-    See :py:meth:`update_geometry_from_asset_footprint` for more details about densification
-    and simplification.
+    See :py:class:`RasterFootprint` for details on densification and
+    simplification.
 
     Args:
-        polygon (Polygon): The input Polygon
-        crs (CRS): The CRS of the input Polygon
-        densification_factor (Optional[int]): The factor by which to increase point density
-            within the polygon. A factor of 2 would double the density of points (placing one
-            new point between each existing pair of points), a factor of 3 would place two points
-            between each point, etc.
-        precision (int): The number of decimal places to include in the coordinates for the
-            reprojected geometry.
-        simplify_tolerance (Optional[float]): All points in the simplified object will be within
-            the tolerance distance of the original geometry, in degrees.
+        polygon (Polygon): The input Polygon.
+        crs (CRS): The CRS of the input Polygon.
+        densification_factor (Optional[int]): The factor by which to
+            increase point density within the polygon before projection to
+            EPSG 4326. A factor of 2 would double the density of points (placing
+            one new point between each existing pair of points), a factor of
+            3 would place two points between each point, etc. Higher
+            densities produce higher fidelity footprints in areas of high
+            projection distortion.
+        precision (int): The number of decimal places to include in the final
+            Polygon vertex coordinates.
+        simplify_tolerance (Optional[float]): Distance, in degrees, within which
+            all locations on the simplified polygon will be to the original
+            polygon.
 
     Returns:
-        Polygon: the reprojected Polygon
+        Polygon: The reprojected Polygon.
     """
     if densification_factor is not None:
-        polygon = Polygon(_densify(polygon.exterior.coords, densification_factor))
+        polygon = Polygon(
+            densify_by_factor(polygon.exterior.coords, factor=densification_factor)
+        )
 
     polygon = shape(transform_geom(crs, "EPSG:4326", polygon, precision=precision))
+    polygon = Polygon([k for k, _ in groupby(polygon.exterior.coords)])
 
     if simplify_tolerance is not None:
         polygon = polygon.simplify(
             tolerance=simplify_tolerance, preserve_topology=False
-        ).simplify(0)
-
-    # simplify does not remove duplicate sequential points, so do that
-    polygon = Polygon([k for k, _ in groupby(polygon.exterior.coords)])
+        )
 
     return polygon
```

Comment:
Ideally, this code could share the same functionality as https://github.com/stac-utils/stactools/pull/396/files#diff-3ec600e1e4dab0a2b3d7aca0a4f8b7054c149fca25e23afbcc7d075c7e0b4e2dR210-R212 ... is that possible?

Reply:
Not clear on your question here...

Reply:
The code in `RasterFootprint.reproject_polygon` (https://github.com/stac-utils/stactools/pull/396/files#diff-3ec600e1e4dab0a2b3d7aca0a4f8b7054c149fca25e23afbcc7d075c7e0b4e2dR303-R310) is the same, functionally, as the code in `densify_reproject_simplify` free function (https://github.com/stac-utils/stactools/pull/396/files#diff-3ec600e1e4dab0a2b3d7aca0a4f8b7054c149fca25e23afbcc7d075c7e0b4e2dR839-R840). Is there any way to only have that logic in one place?

Reply:
Ah. I see. I moved the common logic to a new free function.